### PR TITLE
fix: update proposed chat api

### PR DESCRIPTION
**Pull request by GavinGu07**
Fix the participant breaks triggered by the latest chat participant API update: https://github.com/microsoft/vscode/pull/212148
image

### CODE REVIEW

**Review by tecton — APPROVED** [review]
approve for create command part.

File: packages/vscode-extension/src/chat/utils.ts [hunk 1]
```diff
@@ -50,7 +50,11 @@ export function countMessageTokens(message: LanguageModelChatMessage): number {
     if (!value) {
       continue;
     }
-    numTokens += tokenizer.tokenLength(value);
+    if (key === "role") {
```
Comment (lijie-lee) [inline packages/vscode-extension/src/chat/utils.ts:53]: There is  no need to count token size of `role`.  The `BaseTokensPerMessage` just stands for that. 
> By the way: if wanna count the size of `role`, you should use the real role value like "system", "user" and "assistant"

Comment (lijie-lee) [reply packages/vscode-extension/src/chat/utils.ts:53]: Just notice the ChatParticipant SDK has exposed the [interface](https://github.com/microsoft/vscode/blob/5f78b58b57b7cf84d28d801fed6bb4a48f908601/src/vscode-dts/vscode.proposed.languageModels.d.ts#L166) for counting token size. 

Maybe we could adopt to that later.

Comment (GavinGu07) [reply packages/vscode-extension/src/chat/utils.ts]: Thanks. I've removed the token counting for the `role`. 

`computeTokenLength` will be supported later in another PR.

**Review by lijie-lee — DISMISSED** [review]

**Review by lijie-lee — DISMISSED** [review]
approve the tokenize part

**Review by supkasar — DISMISSED** [review]

**Review by lijie-lee — DISMISSED** [review]

**Review by supkasar — APPROVED** [review]

**Review by 1openwindow — APPROVED** [review]

**Pull request merged**